sklearn/calibration.py

```diff
@@ -658,3 +660,232 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
+        which the last estimator is a classifier.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Binary target values.
+
+    n_bins : int, default=5
+        Number of bins to discretize the [0, 1] interval into when calculating
+        the calibration curve. A bigger number requires more data.
+
+    strategy : {'uniform', 'quantile'}, default='uniform'
+        Strategy used to define the widths of the bins.
+
+        - `'uniform'`: The bins have identical widths.
+        - `'quantile'`: The bins have the same number of samples and depend on
+          predicted probabilities.
+
+    name : str, default=None
+        Name for labeling curve. If `None`, the name of the estimator is used.
+
+    brier_score : bool, default=True
+        If `True`, include Brier score in legend.
+
+    ref_line : bool, default=True
+        If `True`, plots a reference line representing a perfectly calibrated
+        classifier.
+
+    ax : matplotlib axes, default=None
+        Axes object to plot on. If `None`, a new figure and axes is created.
+
+    **kwargs : dict
+        Keyword arguments to be passed to :func:`matplotlib.pyplot.plot`.
+
+    Returns
+    -------
+    display : :class:`~sklearn.metrics.CalibrationDisplay`.
+        Object that stores computed values.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt                     # doctest: +SKIP
+    >>> from sklearn import (datasets, calibration, model_selection,
+    ...                      linear_model)
+    >>> X, y = datasets.make_classification(random_state=0)
+    >>> X_train, X_test, y_train, y_test = model_selection.train_test_split(
+    ...     X, y, random_state=0)
+    >>> clf = linear_model.LogisticRegression(random_state=0)
+    >>> clf.fit(X_train, y_train)
+    LogisticRegression(random_state=0)
+    >>> metrics.plot_calibration_curve(clf, X_test, y_test)  # doctest: +SKIP
+    >>> plt.show()                                           # doctest: +SKIP
+    """
+    check_matplotlib_support("plot_calibration_curve")
+    binary_error = "Only binary classification is supported."
+
+    if not is_classifier(estimator):
+        raise ValueError("The estimator parameter should be a fitted binary "
+                         "classifier")
+
+    predict_proba = getattr(estimator, 'predict_proba', None)
+    decision_function = getattr(estimator, 'decision_function', None)
+    prediction_method = predict_proba or decision_function
+    if prediction_method is None:
+        msg = ("Neither response method 'predict_proba' nor "
+               "'decision_function' are defined in "
+               f"{estimator.__class__.__name__}")
+        raise ValueError(msg)
+
+    y_prob = prediction_method(X)
+
+    if predict_proba is None:
+        y_prob = (y_prob - y_prob.min()) / (y_prob.max() - y_prob.min())
```

Comment:
Should mapping to probability with the logistic function be an option here?

Reply:
I'm not familiar with this area and struggled with this.

* If the estimator has a `predict_proba` and `decision_function` - would the probabilities output by `predict_proba` be best to use (since it is specific to that estimator), instead of transforming the `decision_function` ?
* If the estimator does not have `predict_proba` is it better to use `CalibratedClassifierCV` and use 'sigmoid' (Platt's logistic model)? Of course this was designed for SVMs but wiki says 'it can be applied to other classification models'.
Also are the only estimators that don't have a `predict_proba` method SVM estimators?

Thanks for your insight!

Reply:
Actually, thinking about it `CalibratedClassifierCV` calibrates the probabilities which is different from just transforming the `decision_function` scores (and is computationally expensive) so I can see the benefit of adding logistic function. What do you think about the API? Maybe:
```
decision_score_transform : {'min-max', 'logisitc'}
```

Also:

>   would the probabilities output by `predict_proba` be best to use (since it is specific to that estimator), instead of transforming the `decision_function` ?

is relevant to your note about adding a `response_method` parameter. I decided to opt to hard code preference for `predict_proba` because I thought it might be better than transforming the `decision_function` but I'm not sure about this. Happy to change.